lib/matplotlib/collections.py

```diff
@@ -1132,6 +1132,45 @@ def __init__(self, paths, sizes=None, **kwargs):
     def get_paths(self):
         return self._paths
 
+    def check_consistency(self):
+        """
+        Emit warnings if the lengths of certain properties do not match
+        the number of paths.
+
+        This method checks whether the lengths of `sizes` and `edgecolors`
+        are either 1 or equal to the number of paths in the collection.
+        If not, a warning is issued.
+
+        This is useful for identifying potential bugs or silent mismatches
+        in collection properties when plotting, especially when properties
+        are expected to be applied per-path.
+
+        Notes
+        -----
+        - This method does not raise an error, only a warning.
+        - It is recommended to call this at draw-time to ensure properties
+        have been fully set.
+
+        Warnings
+        --------
+        UserWarning
+            If the number of sizes or edgecolors does not match the number
+            of paths and is not a singleton (length 1).
+
+        Examples
+        --------
+        >>> paths = [Path(...), Path(...), Path(...)]
+        >>> pc = PathCollection(paths, sizes=[10, 20])  # 3 paths, 2 sizes
+        >>> pc.check_consistency()
+        UserWarning: Number of sizes does not match number of paths.
+        """
+        n_paths = len(self.get_paths())
+        if self._sizes is not None and len(self._sizes) not in (1, n_paths):
```

Comment:
```suggestion
        if self._sizes.size not in (0, 1, n_paths):
```
Looking at the `set_sizes` method, we always end up with an array here, though it may be size zero (if None was passed).  I think this is the cause of at least some of the failing tests (any warnings are translated into errors in the tests).